File: app/src/main/java/com/example/todaysshow/adapter/JournalAdapter.java
package com.example.todaysshow.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.todaysshow.object.Journal;
import com.example.todaysshow.R;

import java.util.ArrayList;

public class JournalAdapter extends RecyclerView.Adapter<JournalAdapter.JournalItemView> {
    public ArrayList<Journal> horizontalList;
    public Context context;
    public JournalAdapter(ArrayList<Journal> horizontalList, Context context){
        this.horizontalList = horizontalList;
        this.context = context;
    }

    @NonNull
    @Override
    public JournalItemView onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.journal_list_item, parent, false);
        return new JournalItemView(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull JournalItemView holder, int position) {
        final Journal journal = horizontalList.get(position);
        holder.journal_tv.setText(journal.getJournalStr());
        //holder.journal_iv.setImageResource(journal.getImageResource());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(context, journal.getJournalStr() + " " , Toast.LENGTH_SHORT).show();
            }
        });

    }

    @Override
    public int getItemCount() {
        return horizontalList.size();
    }

    public class JournalItemView extends RecyclerView.ViewHolder {
        TextView journal_tv;
        ImageView journal_iv;

        public JournalItemView(@NonNull View itemView) {
            super(itemView);
            journal_tv = (TextView) itemView.findViewById(R.id.journal_tv);
            journal_iv = (ImageView) itemView.findViewById(R.id.journal_iv);
        }


    }


}

File: app/src/main/java/com/example/todaysshow/adapter/JournalAdapter.kt
package com.example.todaysshow.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.example.todaysshow.ItemClickListener
import com.example.todaysshow.R
import com.example.todaysshow.RecyclerItemClickListener
import com.example.todaysshow.`object`.Journal
import com.example.todaysshow.fragment.CommunityFragment
import com.mikhaellopez.circularimageview.CircularImageView

class JournalAdapter(journals: ArrayList<Journal>, context: Context, itemClickListener: ItemClickListener) : RecyclerView.Adapter<JournalAdapter.JournalHolder>() {
    var journals : ArrayList<Journal> = journals
    var context : Context = context
    var itemClickListener = itemClickListener

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): JournalHolder {
        val inflatedView = LayoutInflater.from(parent.context).inflate(R.layout.journal_list_item, parent, false)
        return JournalAdapter.JournalHolder(inflatedView)
    }

    override fun onBindViewHolder(holder: JournalHolder, position: Int) {
        val journal: Journal = journals.get(position)
        holder.journal_tv.text = journal.getJournalStr()
        holder.journal_iv.setImageResource(journal.getImageResource())
        holder.itemView.setOnClickListener {
            itemClickListener.onItemClicked(holder, journal, position)
        }
    }

    override fun getItemCount(): Int {
        return journals.size
    }


    class JournalHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        var journal_tv = itemView?.findViewById<TextView>(R.id.journal_tv)
        var journal_iv = itemView?.findViewById<CircularImageView>(R.id.journal_iv);
    }
}